api/controllers/v1/ncn.go

```diff
@@ -61,38 +61,71 @@ func NewNcnController(workflowService services.WorkflowService, logger utils.Log
 // @Security  OAuth2Application[admin]
 func (u NcnController) NcnCreateRebuildWorkflow(c *gin.Context) {
 	hostname := c.Param("hostname")
-	err := u.validator.ValidateHostname(hostname)
+	u.createRebuildWorkflow([]string{hostname}, c)
+}
+
+// NcnsCreateRebuildWorkflow
+// @Summary   End to end rolling rebuild ncns (workers only)
+// @Param     include  body  models.CreateRebuildWorkflowRequest  true  "hostnames to include"
+// @Tags      NCNs
+// @Accept    json
+// @Produce   json
+// @Success   200  {object}  models.Workflow
+// @Failure   400  {object}  utils.ResponseError
+// @Failure   404  {object}  utils.ResponseError
+// @Failure   500  {object}  utils.ResponseError
+// @Router    /v1/ncns/rebuild [post]
+// @Security  OAuth2Application[admin]
+func (u NcnController) NcnsCreateRebuildWorkflow(c *gin.Context) {
+	var requestBody models.CreateRebuildWorkflowRequest
+	if err := c.BindJSON(&requestBody); err != nil {
+		errResponse := utils.ResponseError{Message: fmt.Sprint(err)}
+		c.JSON(400, errResponse)
+		return
+	}
+	u.createRebuildWorkflow(requestBody.Hosts, c)
+
+}
+
+func (u NcnController) createRebuildWorkflow(hostnames []string, c *gin.Context) {
+	hostnames = removeDuplicateHostnames(hostnames)
+
+	err := u.validator.ValidateWorkerHostnames(hostnames)
 	if err != nil {
 		errResponse := utils.ResponseError{Message: fmt.Sprint(err)}
 		c.JSON(400, errResponse)
 		return
 	}
-	u.logger.Infof("Hostname: %s", hostname)
-	workflow, err := u.workflowService.CreateRebuildWorkflow(hostname)
+	u.logger.Infof("Hostnames: %v", hostnames)
+
+	workflow, err := u.workflowService.CreateRebuildWorkflow(hostnames)
 
 	if err != nil {
 		errResponse := utils.ResponseError{Message: fmt.Sprint(err)}
 		c.JSON(500, errResponse)
 		return
 	} else {
 		myWorkflow := models.Workflow{
-			Name:      workflow.Name,
-			TargetNcn: workflow.Labels["targetNcn"],
+			Name:       workflow.Name,
+			TargetNcns: []string{workflow.Labels["targetNcn"]}, //todo
```

Comment:
What is this todo for?

Reply:
so this PR is only implement the API controller but not the actual "createWorkflow". it is covered by another ticket. I don't even know what the actual workflow will look like so i put a todo here.